src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -267,7 +287,15 @@ public function startApp(?\yii\log\Logger $logger = null): void
             unset($config['container']);
         }
 
-        $config = $this->mockMailer($config);
+        match ($this->mailMethod) {
+            self::MAIL_CATCH => $config= $this->mockMailer($config),
+            self::MAIL_EVENT_AFTER => $config['components']['mailer']['on ' . BaseMailer::EVENT_AFTER_SEND] = fn(MailEvent $event) => $this->emails[]  = $event->message,
```

Comment:
in these cases it has to be documented, that now user has to make sure that no emails are actually sent out